### PR TITLE
[Menu, Popover, PreviewCard, Select, Tooltip] Force `transition: none` while in `starting-style` status

### PR DESCRIPTION
**Pull request by atomiks**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

This lets the correct transition play while waiting for the correct side to be calculated as the result of collision detection. Animations appear to be unaffected by this.

### CODE REVIEW

**Review by vladmoroz — APPROVED** [review]
@atomiks do we need to remove the animations too?

**Comment by atomiks** [comment]
@vladmoroz I made a demo with animations and they appear to work normally already
https://deploy-preview-992--base-ui.netlify.app/experiments/anchor-side-animations

**Comment by vladmoroz** [comment]
OK!

File: docs/src/app/(private)/experiments/tooltip.tsx [hunk 1]
```diff
@@ -188,6 +188,12 @@ export default function TooltipTransitionExperiment() {
       </Tooltip.Root>
 
       <h3>CSS Transition with `@starting-style`</h3>
+      <p>
+        No longer supported due to{' '}
+        <a href="https://github.com/mui/base-ui/pull/992">
+          https://github.com/mui/base-ui/pull/992
+        </a>
+      </p>
```
Comment (atomiks) [inline docs/src/app/(private)/experiments/tooltip.tsx:196]: `@starting-style` no longer works with this change. Considering it doesn't handle this issue either, it shouldn't be encouraged. This is better for consistency if they always use `[data-starting-style]` and `[data-ending-style]` hooks anyway.

**Pull request merged**